packages/fxa-settings/src/components/PageDeleteAccount/index.tsx

```diff
@@ -0,0 +1,219 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useCallback, useState, ChangeEvent } from 'react';
+import { useForm } from 'react-hook-form';
+import { RouteComponentProps, useNavigate } from '@reach/router';
+import { useAccountDestroyer } from '../../lib/auth';
+import { sessionToken } from '../../lib/cache';
+import { useAlertBar } from '../../lib/hooks';
+import { useAccount } from '../../models';
+import InputPassword from '../InputPassword';
+import FlowContainer from '../FlowContainer';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import AlertBar from '../AlertBar';
+import { HomePath, LockwiseLink, MonitorLink, VPNLink } from '../../constants';
+import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
+import { Checkbox } from '../Checkbox';
+
+type FormData = {
+  password: string;
+};
+
+const checkboxLabels = [
+  'Any paid subscriptions you have will be cancelled',
+  'Any may lose saved information and features within Mozilla products',
+  'Reactivating with this email may not restore your saved information',
+  'Any extensions and themes that you published to addons.mozilla.org will be deleted',
+];
+
+export const PageDeleteAccount = (_: RouteComponentProps) => {
+  usePageViewEvent('settings.delete-account');
+  const { handleSubmit, register, formState, setValue } = useForm<FormData>({
+    mode: 'all',
+    defaultValues: {
+      password: '',
+    },
+  });
+  // TODO: the `.errors.password` clause shouldn't be necessary, but `isValid` isn't updating
+  // properly. I think this is a bug in react-hook-form.
+  const disabled =
+    !formState.isDirty || !formState.isValid || !!formState.errors.password;
+  const [errorText, setErrorText] = useState<string>();
+  const [confirmed, setConfirmed] = useState<boolean>(false);
+  const [accountDeleted, setAccountDeleted] = useState<boolean>(false);
+  const [subtitleText, setSubtitleText] = useState<string>('Step 1 of 2');
+  const [checkedBoxes, setCheckedBoxes] = useState<string[]>([]);
+  const allBoxesChecked = checkboxLabels.every((element) =>
+    checkedBoxes.includes(element)
+  );
+  const navigate = useNavigate();
+  const alertBar = useAlertBar();
+  const goBack = useCallback(() => window.history.back(), []);
+
+  const { primaryEmail } = useAccount();
+
+  const advanceStep = () => {
+    setSubtitleText('Step 2 of 2');
+    setConfirmed(true);
+
+    logViewEvent('flow.settings.account-delete', 'terms-checked.success');
+  };
+
+  const deleteAccount = useAccountDestroyer({
+    onSuccess: () => {
+      setAccountDeleted(true);
+
+      logViewEvent('flow.settings.account-delete', 'confirm-password.success');
```

Comment:
In the current settings page, when a user deletes their account, they are redirected to the email first login page. 

I think this is the correct behavior vs showing an "Account deleted" message, since they can't really do anything in the settings page. It is probably ok to fix in a follow up if it adds too much complexity to this PR.

Reply:
I originally had a redirect, but I thought it may not be the best for accessibility. Seems like a a question for @LZoog

Reply:
If there's text that should be read like a confirmation message, yeah, a redirect isn't a11y friendly since a user might not see the message that the account was successfully deleted.

When you delete an account through old settings, it redirects you to the login page with the message "Account deleted successfully" there. That's totally acceptable since the user can see the message _after_ the redirect. Can we replicate similar behavior?

Reply:
Yeah that sounds good to me 👍

Reply:
Thanks @LZoog, After looking into how we show the "Account Deleted" message in content server after the redirect, I'm afraid we aren't going to be able to do it the same way. In the old settings we call `this.navigate` and pass the template through the backbone router in order to show the message on the signup page.

https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/views/settings/delete_account.js#L185

I'm thinking a workable solution would be to show the message "Your account has been deleted", as we currently are, but have a button to redirect to the `/` path. Though I'm not sure what the button copy should be 🤔 

![](https://user-images.githubusercontent.com/1844554/104383393-c5fbba80-54fd-11eb-953d-b079f596011c.png)

Reply:
> I'm thinking a workable solution would be to show the message

You could also pass a flag via query params to the backbone page and let it display the message.

Reply:
That's a simple solution, I like that, I will add that so we can get this merged.